Qt6::Gui no longer finds GuiPrivate in Qt 6.10 and it needs to be
done manually.

For convenience, not patching the earlier "find_package(Qt6 ...)" so
that the Qt version doesn't prevent applying the patch across versions
(just a quick fix, not meant for upstream as-is).

Note: unsure if upstream's workflow changed but, given past handling,
upstream will likely only fix this when they bump Qt in the appimages
and then bump the required version to >=6.10 rather than do conditions
(may need to change the fix if 6.10 isn't quite ready for wider usage)
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -115,2 +115,6 @@
 
+if (Qt6_VERSION VERSION_GREATER_EQUAL 6.10)
+  find_package(Qt6 COMPONENTS GuiPrivate REQUIRED)
+endif()
+
 # Demangler for the debugger.
--- a/pcsx2-qt/CMakeLists.txt
+++ b/pcsx2-qt/CMakeLists.txt
@@ -256,3 +256,2 @@
 target_include_directories(pcsx2-qt PRIVATE
-	${Qt6Gui_PRIVATE_INCLUDE_DIRS}
 	"${CMAKE_BINARY_DIR}/common/include"
@@ -267,2 +266,3 @@
 	Qt6::Gui
+	Qt6::GuiPrivate
 	Qt6::Widgets
